cms/djangoapps/contentstore/views/videos.py

```diff
@@ -236,6 +256,108 @@ def video_images_handler(request, course_key_string, edx_video_id=None):
+):
+    """
+    Validate 3rd Party Transcription Preferences.
+
+    Arguments:
+        provider: Transcription provider
+        cielo24_fidelity:  Cielo24 transcription fidelity.
+        cielo24_turnaround: Cielo24 transcription turnaround.
+        three_play_turnaround: 3PlayMedia transcription turnaround.
+        preferred_languages: list of language codes.
+
+    Returns:
+        validated preferences or a validation error.
+    """
+    error, preferences = None, {}
+
+    # validate transcription providers
+    transcription_plans = get_3rd_party_transcription_plans()
+    if provider in transcription_plans.keys():
+
+        # Further validations for providers
+        if provider == TranscriptProvider.CIELO24:
+
+            # Validate transcription fidelity
+            supported_fidelities = transcription_plans[provider]['fidelity']
+            if cielo24_fidelity in supported_fidelities.keys():
+
+                # Validate transcription turnaround
+                supported_turnarounds = transcription_plans[provider]['turnaround']
+                if cielo24_turnaround not in supported_turnarounds.keys():
+                    error = _('Invalid cielo24 turnaround.')
+                    return error, preferences
+
+                # Validate transcription languages
+                supported_languages = transcription_plans[provider]['fidelity'][cielo24_fidelity]['languages']
+                if not len(preferred_languages) or not (set(preferred_languages) <= set(supported_languages.keys())):
+                    error = _('Invalid languages.')
+                    return error, preferences
+
+                # Validated Cielo24 preferences
+                preferences = {
+                    'cielo24_fidelity': cielo24_fidelity,
+                    'cielo24_turnaround': cielo24_turnaround,
+                    'preferred_languages': list(preferred_languages),
+                }
+            else:
+                error = _('Invalid cielo24 fidelity.')
+        elif provider == TranscriptProvider.THREE_PLAY_MEDIA:
+
+            # Validate transcription turnaround
+            supported_turnarounds = transcription_plans[provider]['turnaround']
+            if three_play_turnaround not in supported_turnarounds.keys():
+                error = _('Invalid 3play turnaround.')
+                return error, preferences
+
+            # Validate transcription languages
+            supported_languages = transcription_plans[provider]['languages']
+            if not len(preferred_languages) or not (set(preferred_languages) <= set(supported_languages.keys())):
+                error = _('Invalid languages.')
+                return error, preferences
+
+            # Validated 3PlayMedia preferences
+            preferences = {
+                'three_play_turnaround': three_play_turnaround,
+                'preferred_languages': list(preferred_languages),
+            }
+    else:
+        error = _('Invalid provider.')
```

Comment:
@srpearce These errors are for precautionary measures since the front-end / javascript code can easily be compromised that's why we put these validation on backend to avoid any unwanted data in storage.

Reply:
@srpearce Let me know if you have any other feedback on this PR, Will address all together.

Reply:
Sorry, I'm not clear on that last comment. If these error messages won't be visible to end users, they're OK as-is. Does that help?